src/application-navigation/ApplicationNavigation.jsx

```diff
@@ -0,0 +1,192 @@
+   */
+  disablePromptsForNavigationItems: PropTypes.bool,
+  /**
+   * By default, the ApplicationNavigation component will resolve any rendered NavigationPrompts prior to
+   * communicating logout selection with `onSelectLogout`. If `disablePromptsForLogout` is provided,
+   * no NavigationPrompts are resolved when logout is selected.
+   */
+  disablePromptsForLogout: PropTypes.bool,
+  /**
+   * The Object (or function that returns an Object) that specifies the messages
+   * used to prompt the user when navigation items are selected while NavigationPrompts
+   * are rendered by the ApplicationNavigation content.
+   */
+  navigationPromptResolutionOptions: navigationPromptResolutionOptionsShape,
+};
+
+const ApplicationNavigation = ({
+  children,
+  notifications,
+  titleConfig,
+  extensionItems,
+  onSelectExtensionItem,
+  hero,
+  navigationItems,
+  activeNavigationItemKey,
+  onSelectNavigationItem: propOnSelectNavigationItem,
+  userConfig,
+  onSelectSettings,
+  onSelectHelp,
+  onSelectLogout: propOnSelectLogout,
+  utilityItems,
+  onSelectUtilityItem,
+  onDrawerMenuStateChange,
+  disablePromptsForNavigationItems,
+  disablePromptsForLogout,
+  navigationPromptResolutionOptions,
+}) => {
+  const navigationPromptCheckpointRef = useRef();
+
+  const onSelectNavigationItem = useCallback((selectedItemKey) => {
+    if (disablePromptsForNavigationItems) {
+      propOnSelectNavigationItem(selectedItemKey);
+      return;
+    }
+
+    navigationPromptCheckpointRef.current.resolvePrompts(navigationPromptResolutionOptions).then(() => {
+      propOnSelectNavigationItem(selectedItemKey);
+    }).catch((e) => { if (e) throw e; });
+  }, [disablePromptsForNavigationItems, navigationPromptResolutionOptions, propOnSelectNavigationItem]);
+
+  const onSelectLogout = useCallback(() => {
+    if (disablePromptsForLogout) {
+      propOnSelectLogout();
+      return;
+    }
+
+    navigationPromptCheckpointRef.current.resolvePrompts(navigationPromptResolutionOptions).then(() => {
+      propOnSelectLogout();
+    }).catch((e) => { if (e) throw e; });
+  }, [disablePromptsForLogout, navigationPromptResolutionOptions, propOnSelectLogout]);
+
+  return (
+    <TerraApplicationNavigation
+      hero={hero}
+      notifications={notifications}
+      titleConfig={titleConfig}
+      navigationItems={navigationItems}
+      onSelectNavigationItem={propOnSelectNavigationItem && onSelectNavigationItem}
+      activeNavigationItemKey={activeNavigationItemKey}
+      userConfig={userConfig}
+      extensionItems={extensionItems}
+      onSelectExtensionItem={onSelectExtensionItem}
+      utilityItems={utilityItems}
+      onSelectUtilityItem={onSelectUtilityItem}
+      onSelectSettings={onSelectSettings}
+      onSelectHelp={onSelectHelp}
+      onSelectLogout={propOnSelectLogout && onSelectLogout}
+      onDrawerMenuStateChange={onDrawerMenuStateChange}
+    >
+      <ApplicationLoadingOverlayProvider>
```

Comment:
Is this loader just within the nav container?

Reply:
It covers everything below the header; this is documented here: https://github.com/cerner/terra-application/pull/5/files#diff-487c8d381b9f7f279bd497d2373b0701R34

Reply:
Thanks for the link!